internal/datastore/postgres/revisions.go

```diff
@@ -234,11 +234,20 @@ func parseRevisionDecimal(revisionStr string) (datastore.Revision, error) {
 	}}, nil
 }
 
-func createNewTransaction(ctx context.Context, tx pgx.Tx) (newXID xid8, newSnapshot pgSnapshot, err error) {
+func createNewTransaction(ctx context.Context, tx pgx.Tx, metadata map[string]any) (newXID xid8, newSnapshot pgSnapshot, err error) {
 	ctx, span := tracer.Start(ctx, "createNewTransaction")
 	defer span.End()
 
-	cterr := tx.QueryRow(ctx, createTxn).Scan(&newXID, &newSnapshot)
+	if metadata == nil {
+		metadata = map[string]any{}
```

Comment:
let's create a global var for this so we don't allocate (I'm actually unsure if this allocates, but just in case)